src/hooks/useCurrentMarketId.ts

```diff
@@ -60,6 +62,10 @@ export const useCurrentMarketId = () => {
         setLastViewedMarket(marketId);
         dispatch(setCurrentMarketId(marketId));
         dispatch(closeDialogInTradeBox());
+        if (activeTradeBoxDialog) {
+          // Reopen active trade box dialog with cleared values
+          dispatch(openDialogInTradeBox(activeTradeBoxDialog));
```

Comment:
i think we can just close it -- if close dialog was active in ETH and i switched market, i think it's ok that the we go back to default trade form

currently the ux is kinda broken because the inputs aren't cleared (?) and i'm closing position on a market i don't have positions in

Reply:
<img width="1332" alt="image" src="https://github.com/dydxprotocol/v4-web/assets/9400120/9ba542c0-bf08-4bfc-b2ff-139a323bc47a">

Reply:
oh rip. Hm so I need this change otherwise this [blurb](https://github.com/dydxprotocol/v4-web/pull/407/files#diff-c53bac260aa8de1b9d69b0a4094ee1cbcf279108ae8acc553726d0334a4c398dR46) doesn't work as expected (I think `dispatch` ends up executing before `navigate` completes. Eh actually I guess I could use `state` in `NavigateOptions` to set this 🤔  - lemme try that out

Reply:
ah good catch - updated!


https://github.com/dydxprotocol/v4-web/assets/70078372/2a44b74c-1a7f-4f00-91dd-3a922a035623